modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableMessageGroup.java

```diff
@@ -58,4 +58,19 @@ public class TableMessageGroup {
      * Message type for {@link org.apache.ignite.internal.table.distributed.message.HasDataResponse}.
      */
     public static final int HAS_DATA_RESPONSE = 6;
+
+    /**
+     * Message type for {@link org.apache.ignite.internal.table.distributed.replication.request.ReadOnlySingleRowReplicaRequest}.
+     */
+    public static final short RO_SINGLE_ROW_REPLICA_REQUEST = 7;
+
+    /**
+     * Message type for {@link org.apache.ignite.internal.table.distributed.replication.request.ReadOnlyMultiRowReplicaRequest}.
+     */
+    public static final short RO_MULTI_ROW_REPLICA_REQUEST = 8;
+
+    /**
+     * Message type for {@link org.apache.ignite.internal.table.distributed.replication.request.ReadOnlyScanRetrieveBatchReplicaRequest}.
+     */
+    public static final short RO_SCAN_RETRIEVE_BATCH_REPLICA_REQUEST = 9;
```

Comment:
How are you going to close all cursors?
Within RW transactions it's possible to add 
`closeAllTransactionCursors(request.txId()); `to `processTxCleanupAction` handler and to be honest I've just forgot to do this during finish tx ticket.
However that won't work for RO cursors because they might be created on any replica not only primary one, and cleanup request only goes to primary replica.